components/director/internal/domain/scenarioassignment/converter.go

```diff
@@ -0,0 +1,53 @@
+package scenarioassignment
+
+import (
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/pkg/errors"
+)
+
+func NewConverter() *converter {
+	return &converter{}
+}
+
+type converter struct{}
+
+func (c *converter) FromInputGraphql(in graphql.AutomaticScenarioAssignmentSetInput, tenant string) (model.AutomaticScenarioAssignment, error) {
+	out := model.AutomaticScenarioAssignment{
+		ScenarioName: in.ScenarioName,
+		Tenant:       tenant,
+	}
+
+	if in.Selector != nil {
+		out.Selector = model.LabelSelector{
+			Key: in.Selector.Key,
+		}
+
+		strVal, ok := (in.Selector.Value).(string)
+		if ok {
```

Comment:
This can be expressed this way:
```golang
if !ok {
	return model.AutomaticScenarioAssignment{}, errors.New("value has to be a string")
} 

out.Selector.Value = strVal
```